ros_msft_onnx/src/yolo_box.cpp

```diff
@@ -9,40 +9,67 @@
 #include <algorithm>
 #include <numeric>
 #include <functional>
+#include <fstream>
 
 namespace yolo
 {
-    const int ROW_COUNT = 13;
-    const int COL_COUNT = 13;
-    const int CHANNEL_COUNT = 125;
     const int BOXES_PER_CELL = 5;
     const int BOX_INFO_FEATURE_COUNT = 5;
     const int CLASS_COUNT = 20;
     const float CELL_WIDTH = 32;
     const float CELL_HEIGHT = 32;
-
-    static const std::string labels[CLASS_COUNT] =
-    {
-        "aeroplane", "bicycle", "bird", "boat", "bottle",
-        "bus", "car", "cat", "chair", "cow",
-        "diningtable", "dog", "horse", "motorbike", "person",
-        "pottedplant", "sheep", "sofa", "train", "tvmonitor"
-    };
-
+    const int TENSOR_WIDTH = 416;
+    const int TENSOR_HEIGHT = 416;
+    const std::string TRACKER_TYPE = "yolo";
     const std::string kDefaultLabel = "person";
 
     YoloProcessor::YoloProcessor()
     {
         _normalize = false;
     }
 
-    bool YoloProcessor::init(ros::NodeHandle& nh, ros::NodeHandle& nhPrivate)
+    bool YoloProcessor::init(ros::NodeHandle &nh, ros::NodeHandle &nhPrivate)
```

Comment:
The pre-processing and post-processing looks not very alike between yolo and customvision. Maybe it is easier to create another `customvision.cpp` which encapsulate the details and separate from yolo implementation?

Reply:
(or abstracting the common part.)

Reply:
Well customvision model is based on a yolo implementation. The caveat here is the model has a different input/ouput name and the labels are loaded from a text file which also comes with the downloaded onnx zip file